### PR TITLE
0.2.0 Version

### PR DESCRIPTION
**Pull request by emrekeskinmac**
### Added
- getResource selector is added
- getData selector is added 
- hasData selector is added 
- hasError selector is added 
- isComplete selector is added 
- isSuccessComplete selector is added 
- isErrorComplete selector is added 

### Changed
-  Error prop type changed from "Error" to "any"
### Fixed
- resourceFailed error is handled

### CODE REVIEW

File: src/saga.ts [hunk 1]
```diff
@@ -35,7 +30,17 @@ export function resourceStore(options: ResourceStoreOptions) {
       );
       yield put(resourceSucceeded(action.payload.resourceType, response));
     } catch (e) {
-      yield put(resourceFailed(action.payload.resourceType, e));
+      // tslint:disable-next-line:no-console
+      console.log({ e, action });
```
Comment (handeeadiguzel) [inline src/saga.ts]: console log u silelim mi

**Comment by handeeadiguzel** [comment]
`npm link` ile test ettik mi ?

**Comment by emrekeskinmac** [comment]
> `npm link` ile test ettik mi ?

yes

File: src/saga.ts (hunk 1 above)
Comment (emrekeskinmac) [reply src/saga.ts]: bence gerek yok bu şekilde kalabilir

File: package.json [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+    "prePublishOnly": "npm run build"
   },
   "types": "dist/index",
   "typings": "dist/index",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/oplog/resource-redux.git"
   },
-  "author": "Oplog",
+  "author": "OPLOG Development Team <info@oplog.com.tr”> (https://oplog.com.tr)",
```
Comment (ardadevelioglu) [inline package.json]: Team OPLOG yazalım mı? 😄

File: src/types.ts [hunk 3]
```diff
@@ -10,6 +10,12 @@ export type ResourceStoreState = ({ [key: string]: Resource<any> });
 
 export interface Resource<T> {
   data?: T;
-  error?: Error;
+  error?: any;
   isBusy: boolean;
 }
+
+interface Errors {
```
Comment (ardadevelioglu) [inline src/types.ts]: Singular olması gerekmez mi?

**Pull request merged**